photon-lib/src/main/scala/com/linkedin/photon/ml/optimization/TRON.scala

```diff
@@ -249,6 +249,22 @@ class TRON(
 
     objectiveFunction.cleanupCoefficients(convertedPrevCoefficients)
 
+    // If any coefficients exceed the bounds imposed by the constraintMap, knock them back into range
+    if (constraintMap.isEmpty == false) {
+      constraintMap.foreach { map => 
+        map.keys.foreach { key =>
+          val range = map(key)
+          val currentCoef = finalState.coefficients(key)
+          if (currentCoef < range._1) {
+            finalState.coefficients(key) = range._1
+          }
+          else if (currentCoef > range._2) {
+            finalState.coefficients(key) = range._2
+          }
+        }
+      }
+    }
```

Comment:
I think what you would actually want to do is drop these changes and replace line 228 with:
```
OptimizationUtils.projectCoefficientsToSubspace(coefficients, constraintMap),
```